test/unit/org/apache/cassandra/cql3/validation/entities/TupleTypeTest.java

```diff
@@ -233,5 +258,136 @@ public void testReversedTypeTuple() throws Throwable
         DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mmX", Locale.ENGLISH);
         assertRows(execute("SELECT tdemo FROM %s"), row(tuple( df.parse("2017-02-03 03:05+0000"), "Europe")));
     }
+
+    @Test
+    public void tuplePkReadWrite()
+    {
+        qt().withExamples(1).withShrinkCycles(0).forAll(caseGen()).checkAssert(orFail(testcase -> {
+            TupleType tupleType = testcase.type;
+            createTable("CREATE TABLE %s (id " + toCqlType(tupleType) + ", value int, PRIMARY KEY(id))");
+            SortedMap<ByteBuffer, Integer> map = new TreeMap<>(Comparator.comparing(currentTableMetadata().partitioner::decorateKey));
+            int count = 0;
+            for (ByteBuffer value : testcase.uniqueRows)
+            {
+                map.put(value, count);
+                ByteBuffer[] tupleBuffers = tupleType.split(value);
+
+                execute("INSERT INTO %s (id, value) VALUES (?, ?)", tuple(tupleBuffers), count);
```

Comment:
[javac] /Users/jmeredith/oss/trunk-review/test/unit/org/apache/cassandra/cql3/validation/entities/TupleTypeTest.java:275: warning: non-varargs call of varargs method with inexact argument type for last parameter;
    [javac]                 execute("INSERT INTO %s (id, value) VALUES (?, ?)", tuple(tupleBuffers), count);

(and on L277, 278, 312, 314, 315) with Java11.

Reply:
We can just make sure we cast to `Object[]` right?

Reply:
cast for now.  no more compile warnings other than unsafe casts =(